File: src/main/kotlin/magicbook/gtlitecore/api/advancement/impl/AbstractCriterion.java
package magicbook.gtlitecore.api.advancement.impl;

import magicbook.gtlitecore.api.advancement.IAdvancementCriterion;
import net.minecraft.util.ResourceLocation;
import org.jetbrains.annotations.NotNull;

public abstract class AbstractCriterion implements IAdvancementCriterion
{

    private ResourceLocation id = new ResourceLocation("MISSING");

    @NotNull
    @Override
    public ResourceLocation getId()
    {
        return this.id;
    }

    @Override
    public void setId(ResourceLocation id)
    {
        this.id = id;
    }

    @Override
    public String toString()
    {
        return "AbstractCriterion{id=" + this.id + "}";
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/api/advancement/impl/AbstractCriterion.kt
package gregtechlite.gtlitecore.api.advancement.impl

import gregtechlite.gtlitecore.api.advancement.AdvancementCriterion
import net.minecraft.util.ResourceLocation

abstract class AbstractCriterion : AdvancementCriterion
{
    private var id: ResourceLocation = ResourceLocation("MISSING")

    override fun getId(): ResourceLocation = id

    override fun setId(id: ResourceLocation)
    {
        this.id = id
    }

    override fun toString(): String = "AbstractCriterion{id=$id}"
}